### DISCUSSION TITLE
automate a virtual 2 way with 2 switch zigbee - Haade.fr

### DISCUSSION THREAD

**Discussion by giscus** [haade-administrator/haade.fr#42] [Comments]
# automate a virtual 2 way with 2 switch zigbee - Haade.fr

we will simply automate thanks to blueprint and homeassistant a virtual 2 way between 2 single-way zigbee switches

https://haade.fr/en/blog/automate-a-virtual-come-and-go-with-blueprint-and-homeassistant

**Comment by MBRetrofit** [comment]
Hello, as from latest HA update the error message "has an action that calls an unknown service: switch.turn_unavailable " become present after any HA restart.

This happened because: homeassistant.turn_{{ trigger.to_state.state }} will resolve to homeassistant.turn_unavailable when the entity is unavailable, this probably happens during startup.

Automation calls the service: homeassistant.turn_unavailable which does not exist.

"unavailable" is a valid state for any entity (so is unknown). The way you wrote service template in your automation, allows for calling:

homeassistant.turn_unavailable
homeassistant.turn_unknown
You need to make sure the to_state is either on or off to ensure you are not calling a non-existing service (and thus you automation will fail running).

**Reply by MBRetrofit** [reply]
Maybe add something like:
trigger:
  - platform: state
    not_to:
      - unavailable
      - unknown

**Reply by haade-administrator** [reply]
thank you for this interesting feedback were you able to modify the script by adding an on/off trigger if yes can you share

**Reply by MBRetrofit** [reply]
I added trigger condition after variables in Blueprints 2-way-ha.yaml file.
....
variables:
  switch_1: !input switch_1
  switch_2: !input switch_2
trigger:
  - platform: state
    not_to:
      - unavailable
      - unknown

**Comment by MBRetrofit** [comment]
In the mean time your Blueprint becomes deprecated because of syntax error. I tried repairs, but without succes.
